doc/sflow/sflow_hld.md

```diff
@@ -0,0 +1,456 @@
+# sFlow High Level Design
+### Rev 0.3
+## Table of Contents
+
+## 1. Revision 
+Rev | Rev	Date	| Author	| Change Description
+---------|--------------|-----------|-------------------
+v0.1	 |05/01/2019	|Padmanabhan Narayanan	| Initial version
+v0.2    |05/20/2019  |Padmanabhan Narayanan      | Updated based on internal review comments
+v0.3    |06/11/2019  |Padmanabhan Narayanan      | Update CLIs, remove sflowcfgd 
+
+## 2. Scope
+This document describes the high level design of sFlow in SONiC
+
+## 3. Definitions/Abbreviations
+
+Definitions/Abbreviation|Description
+------------------------|-----------
+SAI| Switch Abstraction Interface
+NOS| Network Operating System
+OID| OBject Identifier
+
+## 4. Overview
+
+sFlow (defined in https://sflow.org/sflow_version_5.txt) is a standard-based sampling technology the meets the key requirements of network traffic monitoring on switches and routers. sFlow uses two types of sampling:
+
+* Statistical packet-based sampling of switched or routed packet flows to provide visibility into network usage and active routes
+* Time-based sampling of interface counters.
+
+The sFlow monitoring system consists of:
+
+ * sFlow Agents that reside in network equipment which gather network traffic and port counters and combines the flow samples and interface counters into sFlow datagrams and forwards them to the sFlow collector at regular intervals over a UDP socket. The datagrams consist of information on, but not limited to, packet header, ingress and egress interfaces, sampling parameters, and interface counters. A single sFlow datagram may contain samples from many flows.
+ * sFlow collectors which receive and analyze the sFlow data.
+
+ sFlow is an industry standard, low cost and scalable technique that enables a single analyzer to provide a network wide view.
+
+## 5. Requirements
+
+sFlow will be implemented in multiple phases:
+
+### **Phase I:**
+
+1. sFlow should be supported on physical interfaces.
+2. sFlow should support 2 sFlow collectors.
+3. sFlow collector IP can be either IPv4 or IPv6.
+4. sFlow collector can be reachable via
+	1. 	Front end port
+	2. 	Management port
+5. Set the default sampling rate to 0, configuring non-default value should enable sFlow and start sampling.
+6. Default sFlow sample size should be set to 128 bytes.
+7. Support sFlow related
+	1. CLI show/config commands
+	2. syslogs
+
+
+### **Phase II:**
+1. sFlow should be supported on portchannel interfaces.

```

Comment:
do you support break-out interfaces? Is there any hsflowd limitation to support breakouts?

Reply:
break-out interfaces are not currently supported.